ibc-testkit/src/testapp/ibc/core/core_ctx.rs

```diff
@@ -42,45 +40,44 @@ where
     type HostClientState = AnyClientState;
     type HostConsensusState = AnyConsensusState;
 
-    fn host_height(&self) -> Result<Height, ContextError> {
-        Ok(Height::new(
-            *self.revision_number.lock(),
-            self.store.current_height(),
-        )?)
+    fn host_height(&self) -> Result<Height, HostError> {
+        Height::new(*self.revision_number.lock(), self.store.current_height())
+            .map_err(|e| HostError::invalid_state(e.to_string()))
     }
 
-    fn host_timestamp(&self) -> Result<Timestamp, ContextError> {
+    fn host_timestamp(&self) -> Result<Timestamp, HostError> {
         let host_height = self.host_height()?;
         let host_cons_state = self.host_consensus_state(&host_height)?;
         Ok(host_cons_state.timestamp())
     }
 
-    fn client_counter(&self) -> Result<u64, ContextError> {
-        Ok(self
-            .client_counter
+    fn client_counter(&self) -> Result<u64, HostError> {
+        self.client_counter
             .get(StoreHeight::Pending, &NextClientSequencePath)
-            .ok_or(ClientError::Other {
-                description: "client counter not found".into(),
-            })?)
+            .ok_or(HostError::missing_state(
+                "missing client counter".to_string(),
+            ))
     }
 
-    fn host_consensus_state(
-        &self,
-        height: &Height,
-    ) -> Result<Self::HostConsensusState, ContextError> {
+    fn host_consensus_state(&self, height: &Height) -> Result<Self::HostConsensusState, HostError> {
         let consensus_states_binding = self.host_consensus_states.lock();
-        Ok(consensus_states_binding
+
+        consensus_states_binding
             .get(&height.revision_height())
             .cloned()
-            .ok_or(ClientError::MissingLocalConsensusState(*height))?)
+            .ok_or(HostError::missing_state(
+                ClientError::MissingLocalConsensusState(*height).to_string(),
+            ))
     }
 
     fn validate_self_client(
         &self,
         client_state_of_host_on_counterparty: Self::HostClientState,
-    ) -> Result<(), ContextError> {
+    ) -> Result<(), HostError> {
         if client_state_of_host_on_counterparty.is_frozen() {
-            return Err(ClientError::UnexpectedStatus(Status::Frozen).into());
+            return Err(HostError::invalid_state(
+                "client unexpectedly frozen".to_string(),
```

Comment:
There is no need to apply `.to_string()` in such cases. Please remove them everywhere we use newly defined `HostError` methods.

```suggestion
                "client unexpectedly frozen".to_string(),
```

Reply:
[c5bb0af](https://github.com/cosmos/ibc-rs/pull/1331/commits/c5bb0afd3ca24e839fbfbe7a117229aae0468c2b)